### DISCUSSION TITLE
Pull request template

### DISCUSSION THREAD

**Discussion by sigveio** [cssnano/cssnano#1174] [General]
Could it be worth having a pull request template to help encourage/guide submitters to follow guidelines and provide relevant information?

I made a [suggestion/example](https://github.com/sigveio/cssnano/blob/9df673bfb1c0af2c7606e438eec9123dc48bf935/.github/pull_request_template.md) but thought I would ask for feedback before submitting a PR

**Comment by ludofischer** [comment]
I think it would be a good idea. I was also thinking the issue template could use some improvement. For example, it says to post screenshots so people end up posting a screenshot of the problematic code, which I think is worse than copy paste as it forces you to re-type it.

**Comment by sigveio** [comment]
Yeah I agree with you that they could also do with some tweaks.

For example using Markdown headings instead of bold for the various headings would be cleaner too.

```
<!-- also mention the version in which it was version (if any) -->
```
Do you know what this is intended to mean (from the bug template)?

**Comment by sigveio** [comment]
Ahh, cool, I see that github supports constructing [issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) using YAML. I would propose that we give that a try when replacing the issue template. Then we can provide clear descriptions to explain the different fields.

I think it could save a lot of time to require a minimal reproducible example for issues. But not everyone knows what this means, or why it's important. So I'm thinking our contributing guidelines could include information about this. And to save on writing the same stuff over and over again, perhaps we could add a bot that posts a standard explanation if we add a certain tag (e.g. "Needs reproduction") to an issue? That workflow would also make it easier to see at a glance which issues need attention, and which ones are awaiting more information from the user.

**Comment by sigveio** [comment]
Note to self... should probably also ask for the output of `npm ls postcss` for all issues. Could indicate if additional plugins they use with postcss depend on an old version, and save some time.
